packages/shared/src/logger.ts

```diff
@@ -0,0 +1,99 @@
+import { DANGEROUS__checkEnvironmentVariable } from "./utils";
+
+const logLevels = {
+  debug: 0,
+  info: 1,
+  silent: 999,
+};
+
+type LogLevel = keyof typeof logLevels;
+
+const getLogLevel = () => {
+  const level = DANGEROUS__checkEnvironmentVariable("UT_LOG_LEVEL") as LogLevel;
```

Comment:
instead of relying on env var here, might be better to accept this as an option when creating an uploadthing client / server

(in conjunction with the new url option in #310)

```ts
const { useUploadThing } = generateReactHelpers({
  url: "...",
  logLevel: "debug" // here **users** could also choose to use env vars if they want as they know what runtime they're using
});

const { UploadButton } = generateComponents({
  url: "...",
  logLevel: "debug" // here **users** could also choose to use env vars if they want as they know what runtime they're using
});
```

(same on the server)

```ts
export const { GET, POST } = createNextRouteHandler({
  router: uploadRouter,
  config: {
    logLevel: "debug",
  },
});
```

Reply:
Ideally I'd like the user to only have to set it in one place. But you're probably right, the environment variable story is really not good.

Reply:
Yup, envs works on the server since process is defined there but client is rough 😅

Reply:
I am realizing now though that the log wrapper in the shared package wouldn't have access to that. I'll need to make this as a class that gets instantiated and you pass the config into :/